src/content/1.7/development/architecture/migration-guide/forms/CRUD-form.md

```diff
@@ -0,0 +1,252 @@
+---
+title: CRUD Form
+weight: 20
+---
+
+# CRUD Form
+
+## Introduction
+
+In PrestaShop, many forms represent an identifiable object data such as Cart, Product, Order, Customer and many others.
+Each identifiable object share common actions such as its creation, update or it's display. To wrap the common behavior is a must and
+identifiable object is here for rescue. It allows to retrieve data for form display and to submit it by using following parts:
+
+* _Form data provider_ - provides default data (when creating object) and stored data (when updating object).
+* _Form builder_ - creates identifiable object form and dispatches hook that allows form modifications.
+* _Form data handler_ - handles submitted form data for both create and update actions.
+* _Form handler_ - handles form (in most cases created by FormBuilder), dispatches hooks for create and update action.
```

Comment:
I can imagine some people thinking "mmm ... why does we need so many classes "just" to persist data into database ?"

Maybe we can answer this question by explaining:
- A CRUD page actually requires several things: create a form, fill it with data, validate the data, persist the data
- We follow here SOLID principles: each of our class does answers 1 of these needs. Moreover if a developer needs to modify 1 of these behavior (for example, gather the default data from an external API) he can modify* only the dedicated class behavior to obtain what he needs, not the whole system

*Later we'll explain (in another docs PR I guess) that overrides work here but we strongly suggest using composition instead

Reply:
In such case I believe we will need to put such explanations on every documentation page - I think the process is self explanatory and when we add more documentations such as **How to override form data handler** or something else then people will understand why we are splitting the behaviors. More over, I think we need some kind of generic place to describe that we're following SOLID principes etc...